server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -285,7 +292,9 @@ void processPendingEvents() throws Exception {
           pendingEventTimeoutSec, event.getEvent());
       return;
     }
-    if (storage.canWrite()) {
+    // storage maybe null if the application cache was cleared already
+    // add event to flush queue, and it will be released
+    if (storage == null || storage.canWrite()) {
```

Comment:
when storage equals `null`, why do we need add it to the FlushQueue?

Reply:
Event should be discarded and memory should be updated, these actions can't be finished here

Reply:
Could we change the from
```
 if (System.currentTimeMillis() - event.getCreateTimeStamp() > pendingEventTimeoutSec * 1000L) {
```
to 
```
 if (System.currentTimeMillis() - event.getCreateTimeStamp() > pendingEventTimeoutSec * 1000L || storage == null) {
```
?
It seems better.

Reply:
Or  we add the code
```
    if (storage == null) {
      if (shuffleServer != null) {
        shuffleServer.getShuffleBufferManager().releaseMemory(
            event.getEvent().getSize(), true, false);
      }
      LOG.error("storage is empty , the event {} is dropped",  event.getEvent());
      return;
    }
```

Reply:
ok, I'll update